main.go

```diff
@@ -188,53 +127,38 @@ func (cacheModel *CacheModel) CloseTarArchive() error {
 		ModTime:  time.Now(),
 	}
 
-	if err := cacheModel.TarWriter.WriteHeader(closingHeader); err != nil {
+	if err := tarWriter.WriteHeader(closingHeader); err != nil {
 		return err
 	}
 
-	if _, err := io.Copy(cacheModel.TarWriter, bytes.NewReader(filePathMapBytes)); err != nil && err != io.EOF {
+	if _, err := io.Copy(tarWriter, bytes.NewReader(filePathMapBytes)); err != nil && err != io.EOF {
 		return err
 	}
 
-	if err := cacheModel.TarWriter.Close(); err != nil {
+	if err := tarWriter.Close(); err != nil {
```

Comment:
I think a `defer func() { .close }` would be better for the Writer / File close.

Reply:
@BirmacherAkos There will be no any extra job for tarwriter, in case when you know that you can close a writer just close it straight away. However, if you are expecting something to happen which block you later on to close the writer then in that cases you need to use defer.